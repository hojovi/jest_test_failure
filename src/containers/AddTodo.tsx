import * as React from 'react';
interface IAddTodoProps {
    dispatch: Function;
}

class AddTodoComponent extends React.Component<IAddTodoProps, {}> {
    render(): JSX.Element {
        /* tslint:disable */
        const styles: any = require("./AddTodo.css");
        /* tslint:enable */

        let input = React.createRef<HTMLInputElement>();

        return (
            <div className={styles.aaa}>
                <form
                    onSubmit={e => {
                        e.preventDefault();
                        if (!input.current.value.trim()) {
                            return;
                        }
                        this.props.dispatch();
                        input.current.value = '';
                    }}>
                    <input ref={input} />
                    <button type="submit">
                        Add Todo
                    </button>
                </form>
            </div>
        )
    }
}

const AddTodo = AddTodoComponent;

export default AddTodo;