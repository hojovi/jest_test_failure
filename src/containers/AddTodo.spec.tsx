import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer';
import AddTodo from './AddTodo';

test("aaaa", () => {
    const componenet = ShallowRenderer.create(
        <AddTodo dispatch={() => {}} />
    );

    let tree = componenet.toJSON();
    expect(tree).toMatchSnapshot();
})