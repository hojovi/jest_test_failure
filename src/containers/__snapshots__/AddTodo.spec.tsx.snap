// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aaaa 1`] = `
<div
  className="aaa"
>
  <form
    onSubmit={[Function]}
  >
    <input />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</div>
`;
